spatialmath/box.go

```diff
@@ -352,3 +355,84 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// TODO: add function description
+func (b *box) ToPointCloud(options map[string]interface{}) ([]r3.Vector, error) {
+
+	dims := [3]float64{b.halfSize[0] * 2, b.halfSize[1] * 2, b.halfSize[2] * 2}
+
+	var faces [][]float64
+	// which faces are these
+	for i := 0.0; i <= dims[0]; i += options["xIter"].(float64) {
```

Comment:
starting at the center, adding 2 points per iteration, per face, and working inside to outsid iterating until `i <= halfSize` will mean that we'll run into symmetric errors if the resolution does not evenly divide the dimension

Reply:
agreed, but if we allow the used to dictate the resolution should it be on them to figure this out?
What do you recommend?

Reply:
also suppose we want to render a cube with side length 1.
both 1 / 0.0.3 and 1 / 0.13 would result in symmetric errors with the former being much less noticeable.

Reply:
done